source/clear-linux/tutorials/dlaas-benchmarks.rst

```diff
@@ -0,0 +1,186 @@
+
+The benchmark workload will run in a Kubernetes (k8s) container. We will use Kubeflow and deploy three nodes for this tutorial, to get a decent return.
+
+k8s setup
++++++++++
+
+Please follow the instructions in the :ref:`kubernetes` tutorial to get set up on |CL|.  The k8s community also has `instructions`_ if you are using Ubuntu instead of |CL|.
+
+
+k8s networking
+++++++++++++++
+
+We have used `flannel`_ as the network provider for these tests. If you are comfortable with another network layer, there is nothing that will prevent you from using that.
+Refer to the Kubernetes `networking documentation`_ for setup.
+
+
+Images
+++++++
+
+We will need to create a docker image that will include launcher.py to include the |CL| Deep Learning Stack, and put the benchmarks repo in the right location. From the docker image, run the following
+
+.. code-block:: bash
+
+  mkdir -p /opt
+  git clone https://github.com/tensorflow/benchmarks.git /opt/tf-benchmarks
+  cp launcher.py /opt
+  chmod u+x /opt/*
+
+Your entry point then becomes "/opt/launcher.py".
+
+
+This will build an image which can be consumed directly by TFJob from kubeflow.  We are working to create these images as part of our release cycle.
+
+
+Ksonnet
++++++++
+
+Kubeflow uses Ksonnet to manage deployments, so we need to install that before setting up Kubeflow. On |CL|, follow these steps:
+
+.. code-block:: bash
+
+  swupd bundle-add go-basic-dev
+  export GOPATH=$HOME/go
+  export PATH=$PATH:$GOPATH/bin
+  go get github.com/ksonnet/ksonnet
+  cd $GOPATH/src/github.com/ksonnet/ksonnet
+  make install
+
+
+.. For other OS, follow steps in: https://www.kubeflow.org/docs/guides/components/ksonnet/ .
+
+After the ksonnet installation is complete, ensure that binary `ks` is accessible across the environment.
+
+
+Kubeflow
+========
+Once you have k8s running on your nodes, you can setup Kubeflow by following these instructions from their `quick start guide`_.
+
+.. code-block:: bash
+
+  export KUBEFLOW_SRC=$HOME/kflow
+  export KUBEFLOW_TAG=”v0.3.2”
+  export KFAPP=”kflow_app”
+  export K8S_NAMESPACE=”kubeflow”
+  mkdir ${KUBEFLOW_SRC}
+  cd ${KUBEFLOW_SRC}
+  curl https://raw.githubusercontent.com/kubeflow/kubeflow/${KUBEFLOW_TAG}/scripts/download.sh | bash
+  ${KUBEFLOW_SRC}/scripts/kfctl.sh init ${KFAPP} --platform none
+  cd ${KFAPP}
+  ${KUBEFLOW_SRC}/scripts/kfctl.sh generate k8s
+
+Now you have all the required kubeflow packages, and you can deploy the primary one for our purposes: tf-job-operator.
+
+.. code-block:: bash
+
+  kubectl create namespace ${K8S_NAMESPACE}
+  ks env add default --namespace "${K8S_NAMESPACE}"
+  ks apply default -c tf-job-operator
+
+This creates the CustomResourceDefinition(CRD) endpoint to launch a TFJob.
+
+Running the Deep Learning as a Service TFJob
+++++++++++++++++++++++++++++++++++++++++++++
+
+The jsonnet template files for ResNet50 and Alexnet are available in the |CL| Deep Learning Stack repository. Download and copy these files into
```

Comment:
This step is confusing.  What am I supposed to be doing?  Copying the jsonnet template files from where? into the path defined below.